### PR TITLE
Fix `/autolink test` command for disabled links

### LINKED ISSUE #119: /autolink test seems to ignore disabled links
From what I can see (using 1.2.0 and mattermost 5.23.0), using /autolink test Foo some text just replies "Foo: no change" even when the text should match... Enablink the link and running the command again then gives the expected output.  But that defeats the whole purpos of disabling a link for testing...

### PR DESCRIPTION
**Pull request by levb**
Fixes #119 

Also adds a "test library" of use-case specific links

### CODE REVIEW

**Comment by levb** [comment]
BTW, @hanzei when I run `make all` locally I get lint errors, do I need to do something in my local lint config to fix it and if so should it be checked in?
image

File: server/autolink/lib_jira_test.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
+package autolink_test
+
+import (
+	"testing"
+
+	"github.com/mattermost/mattermost-plugin-autolink/server/autolink"
+)
+
+var jiraTests = []linkTest{
```
Comment (hanzei) [inline server/autolink/lib_jira_test.go:9]: I don't see the need to put this into a global variable. No other tests needs it.

File: server/autolinkclient/client.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ func NewClientPlugin(api PluginAPI) *Client {
 
 func (c *Client) Add(links ...autolink.Autolink) error {
 	for _, link := range links {
-		linkBytes, err := json.Marshal(&link)
+		linkBytes, err := json.Marshal(link)
```
Comment (hanzei) [inline server/autolinkclient/client.go:43]: Good catch

File: server/autolink/lib_jira_test.go (hunk 1 above)
Comment (levb) [reply server/autolink/lib_jira_test.go:9]: I just want to move the relevant template definitions to the top of the file, as de-facto documentation. Eventually - should be exposed as a bunch of godocs? (Orthogonal to any other efforts to "templatize" autolink deployments)

File: server/autolinkclient/client.go (hunk 2 above)
Comment (levb) [reply server/autolinkclient/client.go:43]: It's a lint-only issue, I think - same JSON, but was an annoyance.

**Comment by levb** [comment]
@aaronrothschild this was FYI  as the beginnings of a template lib.

File: server/autolink/lib_jira_test.go (hunk 1 above)
Comment (hanzei) [reply server/autolink/lib_jira_test.go:9]: I just like to have everything related to a test bundled into the `TestX` method. But at the end, that is personal preference.

> Eventually - should be exposed as a bunch of godocs?

What is the value in having godocs for tests?

**Pull request merged**